### ISSUE TITLE
[false-positive] "Ambiguous class property initialization" too strict

### ISSUE THREAD

**Issue by mfn** [kalessil/phpinspectionsea#111]
```PHP
class Foo {
  protected $prop = 'bar';
}
class Baz extends Foo {
  protected $prop = null;
}
```
In that case `Baz::$prop` is marked with this rule but shouldn't, because it's overriding a parent non-null/non-default (!) initialization.

**Comment by kalessil** [comment]
Thank you for reporting @mfn, this is definitely a false-positive
